### PR TITLE
Remove references and logic of system property besu.docker

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
BesuCommand class (and related unit tests) has quite a bit of deprecated logic which sets various defaults depending on if system property besu.docker is present or not. However, this property is not set in Besu's docker file anymore. Hence, the relevant code needs to be removed.

## Fixed Issue(s)
#785 

Signed-off-by: Usman Saleem &lt;usman@usmans.info&gt;

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 1]
```diff
@@ -1960,42 +1954,19 @@ private String genesisConfig() {
   }
 
   private File genesisFile() {
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1956]: Doesn't seem to be much benefit in wrapping this in a function

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 2]
```diff
@@ -1960,42 +1954,19 @@ private String genesisConfig() {
...
 
   public Path dataDir() {
-    if (isFullInstantiation()) {
-      return standaloneCommands.dataPath.toAbsolutePath();
-    } else if (isDocker) {
-      return Paths.get(DOCKER_DATADIR_LOCATION);
-    } else {
-      return getDefaultBesuDataPath(this);
-    }
+    return standaloneCommands.dataPath.toAbsolutePath();
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:1961]: Same as above, think we can inline this function where it is needed

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 3]
```diff
@@ -991,9 +986,7 @@ void setBesuConfiguration(final BesuConfiguration pluginCommonConfiguration) {
 
   private BesuCommand handleStandaloneCommand() {
     standaloneCommands = new StandaloneCommand();
```
Comment (shemnon) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:988]: Is folding StandaloneCommand back into Besu Command part of this PR or a follow on PR?

Comment (usmansaleem) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java]: I missed that bit, I've added it into same PR.

**Pull request merged**